File: wk4/java/keywords.java
// problem link: https://open.kattis.com/problems/keywords

import java.io.*;
import java.math.*;
import java.util.*;

public class keywords {
    public static void main(String[] args) throws IOException {
        BufferedReader r = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(r.readLine());
        PriorityQueue<String> pq = new PriorityQueue<>();
        while(n-- > 0)
            pq.add(r.readLine().toLowerCase().replace(' ', '-'));
        String p = pq.poll();
        int ret = 1;
        while(!pq.isEmpty()) {
            String c = pq.poll();
            if(!p.equals(c)) {
                p = c;
                ret++;
            }
        }
        System.out.println(ret);
    }
}

File: wk4/cpp/keywords.cpp
// problem link: https://open.kattis.com/problems/keywords

#include <bits/stdc++.h>
using namespace std;

int main() {
    ios::sync_with_stdio(false); cin.tie(0); cout.tie(0);
    
    int n, ret = 0;
    cin >> n;
    string l;
    getline(cin, l);
    unordered_set<string> s;
    while(n--) {
        getline(cin, l);
        for(int i = 0; i < l.size(); i++) {
            l[i] = tolower(l[i]);
            if(l[i] == '-')
                l[i] = ' ';
        }
        if(!s.count(l)) {
            s.insert(l);
            ret++;
        }
    }
    cout << ret;

    return 0;
}
